vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -496,40 +471,29 @@
     failed_when: upgraded_cp_nodes_count.stdout|int != 0
 
   - name: Get control plane machines
-    shell: | 
-            kubectl get machines -n "{{ NAMESPACE }}" -l  cluster.x-k8s.io/control-plane -o json |
+    shell: |
+            kubectl get machines -n "{{ NAMESPACE }}" -l cluster.x-k8s.io/control-plane -o json |
             jq -r '[ .items[] | select(.spec.version == "{{ UPGRADED_K8S_VERSION }}") | .status.nodeRef.name ]'
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: new_control_plane_nodes
+    failed_when: >
+      (new_control_plane_nodes.stdout == "[]") or
+      (new_control_plane_nodes.stderr != "") or
+      (new_control_plane_nodes.rc > 1)
```

Comment:
Why should we fail here when there is more than one KCP nodes ?

Reply:
@fmuyassarov oh, we should fail when `new_control_plane_nodes.rc < 3`, right?

Reply:
If we are expecting to have 3 masters, then yes